.circleci/config.yml

```diff
@@ -0,0 +1,74 @@
+version: 2.1
+orbs:
+  docker: circleci/docker@1.3.0
+jobs:
+  build:
+    docker:
+      - image: xyzsam/gem5-aladdin:latest
+    environment:
+      ALADDIN_HOME: /root/project/src/aladdin
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-gem5-build-cache
+      - run:
+          name: Checkout dependencies
+          command: git submodule update --init --recursive
+      - run:
+          name: Compile
+          command: |
+            mkdir -p build/variables
+            cat \<< EOF > build/variables/X86
+            TARGET_ISA = 'x86'
+            CPU_MODELS = 'AtomicSimpleCPU,O3CPU,TimingSimpleCPU'
+            PROTOCOL = 'MESI_Two_Level_aladdin'
+            EOF
+            python2.7 `which scons` build/X86/gem5.opt PROTOCOL=MESI_Two_Level_aladdin --ignore-style -j2
+      - run:
+          name: Running integration tests
+          command: |
+            python3 src/aladdin/integration-test/common/run_cpu_tests.py
+            python3 src/aladdin/integration-test/common/run_ruby_tests.py
```

Comment:
We can add sweep tests here, but they may quite a long time to finish for 2 cores.
```python
pytest src/aladdin/integration-test/common/run_sweep_tests.py -n 2 --sweep-file=src/aladdin/integration-test/common/machesuite_dma.xe
pytest src/aladdin/integration-test/common/run_sweep_tests.py -n 2 --sweep-file=src/aladdin/integration-test/common/machesuite_cache.xe
```
We can add this later.

Reply:
Let's add these later. We may run into other CI issues with this that will need to be resolved.